src/Sulu/Component/Content/Document/Subscriber/ResourceSegmentSubscriber.php

```diff
@@ -155,7 +161,9 @@ public function handlePersistDocument(PersistEvent $event)
         }
 
         $property = $this->getResourceSegmentProperty($document);
-        $this->persistDocument($document, $property);
+        if ($property) {
+            $this->persistDocument($document, $property);
```

Comment:
The problem here is that the `ResourceSegmentBehavior` actually forces the document to have `StructureBehavior` with a `sulu.rlp` tag, which is already a misconception in my opinion. However, with this if the `sulu.rlp` property is only set if available, and it makes the two behaviors a little bit less coupled.